src/soundvectors.py

```diff
@@ -110,225 +122,418 @@ def fields(cls):
         return dataclasses.fields(cls)
 
 
-clts_feature_hierarchy = [  # CLTS feature domains ordered from least to most specific.
-    "type",
-    "manner",
-    "height",
-    "roundedness",
-    "centrality",
-    "place",
-    "phonation",
-]
+class CLTSDomainHierarchy(enum.Enum):
+    """
+    CLTS feature domains ordered from least to most specific.
+    """
+    type = 1
+    manner = 2
+    height = 3
+    roundedness = 4
+    centrality = 5
+    place = 6
+    phonation = 7
+    aspiration = 8
+    creakiness = 9
+    release = 10
+    ejection = 11
+    pharyngealization = 12
+    voicing = 13
+    nasalization = 14
+    preceding = 15
+    labialization = 16
+    syllabicity = 17
+    palatalization = 18
+    duration = 19
+    stress = 20
+    airstream = 21
+    velarization = 22
+    laminality = 23
+    articulation = 24
+    breathiness = 25
+    glottalization = 26
+    raising = 27
+    relative_articulation = 28
+    rhotacization = 29
+    friction = 30
+    rounding = 31
+    start = 32
+    contour = 33
+    to_roundedness = 34
+
+    @classmethod
+    def primary_domains(cls):
+        return {i for i in itertools.takewhile(lambda i: i.value < 8, cls)}
+
+    @classmethod
+    def exclude_for_to_sound(cls):
+        return {cls.raising, cls.relative_articulation, cls.rounding}
+
+
+@functools.total_ordering
+class HierarchicalFeature:
```

Comment:
"HierarchicalFeature" seemed to be a suitable name for these things - as opposed to the "binary feature" bundles/vectors.